### PR TITLE
E2EE fix upload parallelism issue.

### LINKED ISSUE #1269: Bulk sync of encrypted files is stuck
### Expected behaviour
All encrypted files should be synced.

### Actual behaviour
The first file is correctly synced, but then the sync is stuck.
The encrypted folder is shown as 0 B.
Restarting the sync or the nextcloud client does not help, it causes one more file to sync and then it is stuck again.

### Steps to reproduce
1. Create a new empty folder.
2. Encrypt the folder.
3. Quickly generate 100 files in that folder.
`for i in {1..100}; do dd if=/dev/urandom of=sample_$i bs=100M count=1; done`

(Expected behavior can be produced using by adding a timeout, eg `for i in {1..100}; do dd if=/dev/urandom of=sample_$i bs=100M count=1; sleep 30; done`)

### Client configuration
Client version:
2.5.2git

Operating system:
Fedora Workstation 29

OS language:
English US

Qt version used by client package (Linux only, see also Settings dialog):
5.11.0

Client package (From Nextcloud or distro) (Linux only):
Nextcloud-2.5.2-x86_64.AppImage

Installation path of client:

### Server configuration
Operating system:
CentOS 7

Web server:
Apache 2.4.6

Database:
MariaDB 10.2.24

PHP version:
7.2.18

Nextcloud version:
15.0.7

Storage backend (external storage):

### Logs
1. Client logfile:
https://gist.github.com/Linuxfabrik/fe12ac5d2358d0f1ae4e0c4002594708

2. Web server error log:
Nothing related.

3. Server logfile: nextcloud log (data/nextcloud.log):
Nothing related.

# Possibly related issue
https://github.com/nextcloud/desktop/issues/890
https://github.com/nextcloud/desktop/issues/1168

### LINKED ISSUE #2765: Syncing e2e folders gets stuck
### Expected behaviour
The problem is probably the same as in #890, so expected / actual behavior is the same as in that issue. 
As I couldn't reopen that issue, I am opening a new one.

The workaround / solution includes removing e2e locks in the database on the server, so maybe a separate issue should be filed in NC server ?

(continuing from #890)
For me the problem appeared on NC client versions 2.6.5 (latest in ubuntu bionic ppa),
and Nextcloud-3.0.3-x86_64.AppImage Nextcloud-3.1.0-x86_64.AppImage app images.

There seems to be other users for which it appears also.

Problem (repeating the description):

The sync of an e2e encrypted folder  gets stuck in the client. 
No matter what you do (force sync, cancel sync, restart client, ...) it remains stuck.

Solution:
you need to clean file locks and e2e locks on the server.
Bash:
```bash
DB_PREFIX="oc"
# delete all locked files and all e2e encryption locks
mysql --defaults-extra-file="${HOME}/mysql_options.cnf" -e "DELETE FROM ${DB_PREFIX}_file_locks WHERE 1;"
mysql --defaults-extra-file="${HOME}/mysql_options.cnf" -e "DELETE FROM ${DB_PREFIX}_e2e_encryption_lock;"
```

with `mysql_options.cnf` having the options to connect to the NC databse:
```
[client]
password='???'
user='???'
database='???'
```

For me, the sync started to work again after clearing the e2e encryption locks table.
On the server side, there seems to be some connection issues between
the database and nc host, with connection dropping somewhat frequently
(visible in the NC admin logs).

Questions for the developers:
a) I read in NC support forum that the file locks table is cleared every 30 minutes through
a cron job. Is the same done for the e2e locks ? In case it is not, this may
explain why e2e locks cause problems more frequently.

b) When using memcache (Redis?): are e2e locks stored in memcache, or only file locks
are in memcahce ? This might also explain the difference in occurrences of both problems.

For me, I am now running the bash script on the server via cron.

### PR DESCRIPTION
**Pull request by allexzander**
Signed-off-by: allexzander <blackslayer4@gmail.com>

Fix #2765 
Fix #1269

### CODE REVIEW

File: src/libsync/propagateremotemkdir.cpp [hunk 1]
```diff
@@ -34,29 +34,18 @@ PropagateRemoteMkdir::PropagateRemoteMkdir(OwncloudPropagator *propagator, const
...
-        }
-    }();
     const auto path = _item->_file;
     const auto slashPosition = path.lastIndexOf('/');
     const auto parentPath = slashPosition >= 0 ? path.left(slashPosition) : QString();
 
     SyncJournalFileRecord parentRec;
     bool ok = propagator->_journal->getFileRecord(parentPath, &parentRec);
     if (!ok) {
+        // TODO: not sure how this can be possible, maybe this job should not get created if the parent folder is missing?
```
Comment (er-vin) [inline src/libsync/propagateremotemkdir.cpp:44]: This is more a problem of DB failure. We got a stress test for such cases (testdatabaseerror.cpp). This pushes us to handle this kind of cases as gracefully as possible, in this case that means returning early and maybe have the sync fail later.

**Review by er-vin — COMMENTED** [review]
Lovely, just please remove those TODO comments. Nothing needs being done there AFAICT, this is DB error handling.

**Comment by allexzander** [comment]
@er-vin All done.

**Review by er-vin — APPROVED** [review]

**Pull request merged**